creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -377,37 +436,139 @@ class SnapbackSM {
+      unhealthyReplicasSet,
+      primary,
+      secondary1,
+      secondary2,
+      wallet,
+      secondariesToClockMap
+    })
+
+    // If the primary is still in the unhealthy replica set, and this primary for this wallet has not
+    // been visited before, add to in memory map and skip reconfig.
+    if (unhealthyReplicasSet.has(primary) && !this.walletInUnhealthyPrimaryMap(primary, wallet)) {
+      this.addWalletToUnhealthyPrimaryMap(primary, wallet)
+      response.issueReconfig = this.isModeEnabled(reconfigModes.RECONFIG_DISABLED.key)
+      return response
+    }
+
+    // Note: for v0, only issue reconfigs if only 1 secondary is unhealthy. Else, just log out the tentative
+    // new replica set to be assigned.
+
+    let newPrimary
+    if (unhealthyReplicasSet.size === 1) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If snapbackSM has already checked this primary and it failed the health check, select the higher clock
+        // value of the two secondaries as the new primary, leave the other as the first secondary, and select a new second secondary
+        let currentHealthySecondary
+        ([newPrimary, currentHealthySecondary] = secondariesToClockMap[secondary1] >= secondariesToClockMap[secondary2] ? [secondary1, secondary2] : [secondary2, secondary1])
+        response.newPrimary = newPrimary
+        response.newSecondary1 = currentHealthySecondary
+        response.newSecondary2 = newReplicaNodes[0]
+        response.issueReconfig = this.isModeEnabled(reconfigModes.PRIMARY_AND_OR_SECONDARIES.key)
+      } else {
+        // If one secondary is unhealthy, select a new secondary
+        const currentHealthySecondary = !unhealthyReplicasSet.has(secondary1) ? secondary1 : secondary2
+        response.newPrimary = primary
+        response.newSecondary1 = currentHealthySecondary
+        response.newSecondary2 = newReplicaNodes[0]
+        response.issueReconfig = this.isModeEnabled(reconfigModes.ONE_SECONDARY.key)
+      }
+    } else if (unhealthyReplicasSet.size === 2) {
+      if (unhealthyReplicasSet.has(primary)) {
+        // If primary + secondary is unhealthy, use other healthy secondary as primary and 2 random secondaries
+        response.newPrimary = !unhealthyReplicasSet.has(secondary1) ? secondary1 : secondary2
+        response.issueReconfig = this.isModeEnabled(reconfigModes.PRIMARY_AND_OR_SECONDARIES.key)
+      } else {
+        // If both secondaries are unhealthy, keep original primary and select two random secondaries
+        response.newPrimary = primary
+        response.issueReconfig = this.isModeEnabled(reconfigModes.MULTIPLE_SECONDARIES.key)
+      }
+      response.newSecondary1 = newReplicaNodes[0]
+      response.newSecondary2 = newReplicaNodes[1]
+    }
+
+    // If code reached this point, new nodes have been selected for reconfig. Remove the primary-wallet from map.
+    this.removeWalletFromUnhealthyPrimaryMap(primary, wallet)
```

Comment:
shouldn't this logic also be nested under the check of this node is not primary?

Reply:
eh this is fine, resolving